pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -622,24 +640,33 @@ private void searchForCandidateBroker(NamespaceBundle bundle,
     }
 
     private boolean isBrokerActive(String candidateBroker) {
-        List<String> brokers = pulsar.getLocalMetadataStore().getChildren(LoadManager.LOADBALANCE_BROKERS_ROOT).join();
-
-        for (String brokerHostPort : brokers) {
-            if (candidateBroker.equals("http://" + brokerHostPort)) {
+        URI uri = URI.create(candidateBroker);
```

Comment:
URI os generally expensive. Do we need this or it is worth to do it another way?

Reply:
It's useful since the input is an URL (also an URI). I checked the implementation of URI parsing and it seems very efficient. If this becomes a bottleneck, it could be cached.

Reply:
Actually good point @eolivelli . I'll modify it to search for "://" and just take the remaining characters.

Reply:
Done. I also fixed the other location there URI.create was used.